### PR TITLE
Fix porfolio_reflection list retreival

### PR DESCRIPTION
**Pull request by talhaaslam01**
### Portfolio listing combined with reflection [GP-546](https://edlyio.atlassian.net/jira/software/projects/GP/boards/48?assignee=624d4ad5f4079800705a35e8&selectedIssue=GP-546), [GP-515](https://edlyio.atlassian.net/jira/software/projects/GP/boards/48?assignee=624d4ad5f4079800705a35e8&selectedIssue=GP-515)

### CODE REVIEW

File: openedx/features/genplus_features/genplus_teach/api/v1/views.py [hunk 1]
```diff
@@ -333,3 +334,42 @@ def list(self, request, *args, **kwargs):
...
+            query = Q1 & (Q3 | Q4)
+            
+            if skill:
+                query &= Q7
+            
+            if gtcs:
+                query &= Q8
+        else:
+            query = (Q1 & (Q3 | Q4)) | (Q2 & (Q5 | Q6))
+
```
Comment (hasnain-saeed) [inline openedx/features/genplus_features/genplus_teach/api/v1/views.py:373]: @talhaaslam01 can we change it to something like this and completely remove Q variables?
```
query = Q(portfolio__teacher=teacher) & (Q(portfolio__title__icontains=search) | Q(portfolio__description__icontains=search))

if skill:
    query &= Q(portfolio__skill__id=skill)
if gtcs:
    query &=  Q(portfolio__gtcs__id=gtcs)

if not (skill or gtcs):
    query |= (Q(reflection__teacher=teacher) & (Q(reflection__answer__icontains=search) | Q(reflection__reflection__title__icontains=search)))
```
Also it isn't clear why reflection filters relate to the absence of both skill and gtcs.

**Review by hasnain-saeed — CHANGES_REQUESTED** [review]
Left a comment in PortfolioReflectionView

File: openedx/features/genplus_features/genplus_teach/api/v1/views.py (hunk 1 above)
Comment (talhaaslam01) [reply openedx/features/genplus_features/genplus_teach/api/v1/views.py:373]: @hasnain-saeed Reflection does not have skill and gtcs fields

Comment (talhaaslam01) [reply openedx/features/genplus_features/genplus_teach/api/v1/views.py:372]: `if not (skill or gtcs)` If this conditioned is removed then, the reflections are fetched regardless of the applied filters

**Review by hasnain-saeed — APPROVED** [review]
LGTM 👍 please squash and merge

**Pull request merged**